File: src/main/java/com/trabalho_testes/trabalho_testes/entidade/Acao.java
package com.trabalho_testes.trabalho_testes.entidade;

import lombok.Data;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.List;

@Entity
@Data
public class Acao {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nome;

    private BigDecimal pontos;

    @OneToMany(mappedBy = "acao")
    private List<AcaoJogador> listaAcaoJogador;
}

File: frontend/src/app/app/servico/acao/acao.ts
import { IAcaoJogador, AcaoJogador } from '../acao-jogador/acao-jogador';

export class IAcao {
    id: number;
    total: number;
    listaAcaoJogador: IAcaoJogador[];
}

export class Acao {
    id: number;
    nome: string;
    pontos: number;
    listaAcaoJogador: AcaoJogador[];
}